src/toast/util/toastReducer.ts

```diff
@@ -12,13 +12,20 @@ function toastReducer(state: ToastState, action: ToastAction): ToastState {
   switch (action.type) {
     case "DISPLAY": {
       const {toastData} = action;
+      const {limit} = state;
+      const toastStack = [
+        ...state.toastStack.filter(not(isSameToast(toastData.id))),
+        toastData
+      ];
+
+      if (limit && toastStack.length > limit) {
+        // pop first toast if stack exceeds limit
+        toastStack.shift();
```

Comment:
I'd prefer to do it with `slice` because `shift` directly mutates the `toastStack`